metricflow/engine/metricflow_engine.py

```diff
@@ -144,6 +144,15 @@ def rendered_sql(self) -> SqlQuery:
 
         return sql_query
 
+    @property
+    def rendered_sql_without_descriptions(self) -> SqlQuery:
+        """Return the SQL query without the inline descriptions."""
+        sql_query = self.rendered_sql
+        return SqlQuery(
+            sql_query="\n".join(filter(lambda line: "--" not in line, sql_query.sql_query.split("\n"))),
```

Comment:
Careful. This:

```
SELECT
  ,metric_time  -- ds or date, depending on the measure
  ,measure_one
  ...
```

Produces this:

```
SELECT
  , measure_one
  ...
```

You should test it, but I think this works better, although it'll leave end-of-line comments:

```suggestion
            sql_query="\n".join(filter(lambda line: line.lstrip().startswith("--"), sql_query.sql_query.split("\n"))),
```

Reply:
Note we could try to get fancy with string splitting or other detection to remove end-of-line comments as well, but we'd have to be careful about this:

```
constraint: 'user_flags NOT IN ("--help", "-h")'
```
SQL is not a regular language, which means getting that right requires the AST, so I'd go with the simpler startswith approach. It should be good enough, anyway.